packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -379,6 +587,10 @@ - (void)isInfoWindowShownForMarkerWithIdentifier:(NSString *)identifier
   }
 }
 
+- (CGFloat)getScreenScale {
+  return self.mapView.traitCollection.displayScale;
```

Comment:
@hellohuanlin is this acceptable fix for getting screenScale for now. See comment here: https://github.com/flutter/packages/pull/4055#discussion_r1568646976
This is currently only open review issue before splitting this tasks to sub PR's

Reply:
@stuartmorgan this has been waiting to be resolved last few weeks?
I would like to start plitting this to sub-pr's but this one issue is still blocking the process. Any way to have somekind of solution for this?
This is a bit bigger issue as the markers are initialized before the virw is attached to the parent view. The whole maps initialialization should be changed a bit. Or platformview factory should be able to get some info about the screen where it will be attached to, but that is not cyrrently possible (if the screen scale is not sent in platform view creation params from the dart implementation).

Reply:
I would file an issue explaining the issue, add a comment on this line with a TODO pointing to the issue, and proceed with that. As you pointed out, the cases where this won't work correctly should be uncommon, and likely require Flutter SDK-level changes to correctly handle.